src/lqcontrol.jl

```diff
@@ -107,81 +83,103 @@ function update_values!(lq::LQ)
     # simplify notation
-    Q, R, A, B, C = lq.Q, lq.R, lq.A, lq.B, lq.C
+    Q, R, A, B, N, C = lq.Q, lq.R, lq.A, lq.B, lq.N, lq.C
 
     # solve Riccati equation, obtain P
-    A0, B0 = sqrt(lq.bet) .* A, sqrt(lq.bet) .* B
-    P = solve_discrete_riccati(A0, B0, R, Q)
+    A0, B0 = sqrt(lq.bet) * A, sqrt(lq.bet) * B
+    P = solve_discrete_riccati(A0, B0, R, Q, N)
 
     # Compute F
-    S1 = Q .+ lq.bet .* (B' * P * B)
-    S2 = lq.bet .* (B' * P * A)
-    F = S1 \ S2
+    s1 = Q + lq.bet * (B' * P * B)
+    s2 = lq.bet * (B' * P * A) + N
+    F = s1 \ s2
 
     # Compute d
-    d = lq.bet .* trace(P * C * C') / (1 - lq.bet)
+    d = lq.bet * trace(P * C * C') / (1 - lq.bet)
 
     # Bind states
     lq.P, lq.F, lq.d = P, F, d
-    nothing
 end
 
 function stationary_values(lq::LQ)
-    stationary_values!(lq)
-    return lq.P, lq.F, lq.d
+	_lq = LQ(
+			copy(lq.Q), 
+			copy(lq.R), 
+			copy(lq.A), 
+			copy(lq.B), 
+			copy(lq.C), 
+			copy(lq.N),
+			copy(lq.bet), 
+			lq.term,
+			copy(lq.rf))
+
+    stationary_values!(_lq)
+    return _lq.P, _lq.F, _lq.d
 end
 
+# Dispatch for a scalar problem
+function _compute_sequence{T}(lq::LQ, x0::T, policies)
+	term = length(policies)
+
+	x_path = Array(T, term+1)
+	u_path = Array(T, term)
+	
+	x_path[1] = x0
+	u_path[1] = -(first(policies)*x0)
+	w_path    = lq.C * randn(term+1)
+
+    for t = 2:term
+        f = policies[t]
+        x_path[t] = lq.A*x_path[t-1] + lq.B*u_path[t-1] + w_path[t]
+        u_path[t] = -(f*x_path[t])
+    end
+    x_path[end] = lq.A*x_path[term] + lq.B*u_path[term] + w_path[end]
 
-function compute_sequence(lq::LQ, x0::ScalarOrArray, ts_length=100)
-    # simplify notation
-    Q, R, A, B, C = lq.Q, lq.R, lq.A, lq.B, lq.C
+    x_path, u_path, w_path
+end
 
-    # Preliminaries,
-    if lq.T != nothing
-        # finite horizon case
-        T = min(ts_length, lq.T)
-        lq.P, lq.d = lq.Rf, 0.0
-    else
-        # infinite horizon case
-        T = ts_length
-        stationary_values!(lq)
-    end
+# Dispatch for a vector problem
+function _compute_sequence{T}(lq::LQ, x0::Vector{T}, policies)
+    n, j = size(lq.C,1), size(lq.C,2)
+	term = length(policies)
 
-    # Set up initial condition and arrays to store paths
-    x0 = reshape([x0], lq.n, 1)  # make sure x0 is a column vector
-    x_path = Array(eltype(x0), lq.n, T+1)
-    u_path = Array(eltype(x0), lq.k, T)
-    w_path = C * randn(lq.j, T+1)
+    x_path = Array(T, n, term+1)
+    u_path = Array(T, n, term)
 
-    # Compute and record the sequence of policies
-    policies = Array(typeof(lq.F), T)
+    C = reshape(lq.C,n,j) # Ensure correct dimensionality for w_path
+    w_path = [vec(C*randn(j)) for i=1:(term+1)]
+	
+	x_path[:,1] = x0
+	u_path[:,1] = -(first(policies)*x0)
 
-    for t=1:T
-        if lq.T != nothing
-            update_values!(lq)
-        end
-        policies[t] = lq.F
+    for t = 2:term
+        f = policies[t]
+        x_path[:,t] = lq.A*x_path[t-1] + lq.B*u_path[t-1] + w_path[t]
```

Comment:
@ZacCranko, I thill think there are bugs here. We need to be pulling out `x_path[:, t-1]` and `u_path[:, t-1]`. 

Also, we will still have the issues with dimensionality if we let `lq.B` be a `Vector` and `u_path[:, t-1]` has more than one element.